include/drivers/dac.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief DAC public API header file.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_DAC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_DAC_H_
+
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief DAC driver APIs
+ * @defgroup dac_interface DAC driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/**
+ * @brief Structure for specifying the configuration of a DAC channel.
+ */
+struct dac_channel_cfg {
+	/**
+	 * Channel identifier of the DAC that should be configured.
+	 */
+	u8_t channel_id;
+
+	/**
+	 * Enable buffered output if available in underlying hardware.
+	 */
+	bool buffered;
+};
+
+/**
+ * @brief Type definition of DAC API function for configuring a channel.
+ * See dac_channel_setup() for argument descriptions.
+ */
+typedef int (*dac_api_channel_setup)(struct device *dev,
+				     const struct dac_channel_cfg *channel_cfg);
+
+/**
+ * @brief Type definition of DAC API function for setting a write request.
+ * See dac_write() for argument descriptions.
+ */
+typedef int (*dac_api_write)(struct device *dev,
+			    u8_t channel, u32_t value, u8_t resolution);
+
+/**
+ * @brief DAC driver API
+ *
+ * This is the mandatory API any DAC driver needs to expose.
+ */
+struct dac_driver_api {
+	dac_api_channel_setup channel_setup;
+	dac_api_write         write;
+	u16_t ref_internal;	/* mV */
```

Comment:
I believe this is too simple for a generic API. Something along the lines the ADC API methods for selecting a reference voltage would be ideal. The NXP KE1xF, for instance, supports two external DAC voltage references.

Reply:
Which part of the API should we take from ADC API? I think the reference is as simple in ADC API as here.
Unfortunately, in STM32 the ADC internal reference is shared with DAC reference, so it's not possible to just set the reference voltage in the DAC driver. We might need something like a dedicated reference voltage driver then.

Reply:
The part where the user can select a reference: https://github.com/zephyrproject-rtos/zephyr/blob/1421dfb3f13602189210b590946433ec093c47cd/include/drivers/adc.h#L68

Reply:
Made the reference handling more similar to the ADC now. For the mentioned NXP MCUs we can now select external 0 or 1. For STM32 I added `DAC_REF_DEFAULT`, as I think that `ADC_REF_INTERNAL` as used for the STM32 ADC is a bit misleading. The reference is in most cases not internal, but defined externally (either via VREF+ pin or VDDA). There is no possibility to select a dedicated internal ADC or DAC reference in STM32. It's only possible to drive the VREF+ pin by a buffer for some MCUs (e.g. G4).